src/lib/comms/include/sol-mavlink.h

```diff
@@ -0,0 +1,405 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * @brief Routines to handle Mavlink protocol.
+ *
+ * Wrapper library for Mavlink communication.
+ */
+
+/**
+ * @defgroup MAVLINK MAVLINK
+ * @ingroup Comms
+ *
+ * MAVLink or Micro Air Vehicle Link is a protocol for communicating with
+ * small unmanned vehicle. It is designed as a header-only message marshaling
+ * library.
+ *
+ * @{
+ */
+
+/**
+ * @struct sol_mavlink
+ *
+ * @brief Mavlink Object
+ *
+ * @see sol_mavlink_connect
+ *
+ * This object is the abstraction of a Mavlink connection. This is the base
+ * structure for all Mavlink operations and is obtained through the
+ * sol_mavlink_connect API.
+ */
+struct sol_mavlink;
+
+typedef enum {
+    SOL_MAVLINK_MODE_ACRO = 1,
```

Comment:
not sure if all of these fields require description, but at least a few of these do, don't you think?